cpp/daal/src/algorithms/dtrees/forest/classification/oneapi/df_classification_predict_dense_oneapi_impl.i

```diff
@@ -0,0 +1,495 @@
+/* file: df_classification_predict_dense_oneapi_impl.i */
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+/*
+//++
+//  Implementation of auxiliary functions for decision forest classification
+//  hist method.
+//--
+*/
+
+#ifndef __DF_CLASSIFICATION_PREDICT_DENSE_ONEAPI_IMPL_I__
+#define __DF_CLASSIFICATION_PREDICT_DENSE_ONEAPI_IMPL_I__
+
+#include "src/algorithms/dtrees/forest/classification/oneapi/df_classification_predict_dense_kernel_oneapi.h"
+#include "src/algorithms/dtrees/forest/classification/oneapi/cl_kernels/df_batch_predict_classification_kernels.cl"
+
+#include "src/algorithms/dtrees/forest/classification/df_classification_model_impl.h"
+
+#include "src/externals/service_ittnotify.h"
+#include "services/buffer.h"
+#include "data_management/data/numeric_table.h"
+#include "src/data_management/service_numeric_table.h"
+#include "services/env_detect.h"
+#include "services/error_indexes.h"
+#include "src/services/service_data_utils.h"
+#include "src/services/service_algo_utils.h"
+#include "src/services/service_arrays.h"
+#include "src/services/service_utils.h"
+#include "sycl/internal/types.h"
+
+using namespace daal::services::internal;
+using namespace daal::internal;
+
+namespace daal
+{
+namespace algorithms
+{
+namespace decision_forest
+{
+namespace classification
+{
+namespace prediction
+{
+namespace internal
+{
+inline char * utoa(uint32_t value, char * buffer, size_t buffer_size)
```

Comment:
Please check with @averbukh that this function satisfies SDL. I saw there are restriction about using null-terminated strings

Reply:
thanks, migrated to daal_int_to_string